dashboard/app/models/pd/application/teacher1920_application.rb

```diff
@@ -330,6 +330,28 @@ def friendly_scholarship_status
       end
     end
 
+    def allow_sending_principal_email?
+      response = Pd::Application::PrincipalApproval1920Application.find_by(application_guid: application_guid)
+      principal_approval_email = emails.where(email_type: 'principal_approval').last
+      last_principal_approval_email_sent_at = principal_approval_email&.sent_at
+
+      # Do we allow manually sending/resending the principal email?
+
+      # Only if this teacher application is currently unreviewed or pending.
+      return false unless unreviewed? || pending?
+
+      # Only if the principal approval is required.
+      return false if principal_approval_not_required
+
+      # Only if we haven't gotten a principal response yet.
+      return false if response
+
+      # Only if it's been more than 5 days since we last sent the principal an email.
+      return false if last_principal_approval_email_sent_at && last_principal_approval_email_sent_at > 5.days.ago
```

Comment:
do we ever queue principal approval emails to send in the scheduled pd emails cron job? what should happen here if we have a queued but not sent principal approval email? (`sent_at: nil`) - it looks like this would return `true` and allow us to send another one.

Reply:
Not that I've found yet, but i think I will start examining `created_at` rather than `sent_at`.